src/utils.js

```diff
@@ -209,3 +216,100 @@ export function createModelledFunctionCall(
     }
   };
 }
+
+export function isBindingMutationOutsideFunction(
+  binding: Binding,
+  bindingEntry: BindingEntry,
+  effects: Effects,
+  F: FunctionValue
+): boolean {
+  // If either the previous values are uninitialized then this is not a mutation
+  if (binding.value === undefined || bindingEntry.value === undefined) {
```

Comment:
That's what I thought, but there seems to be cases where it can be defined due to joining of bindings where the binding never existed in the other set of effects.

Reply:
Huh, I can see that being the case, but we should probably drop a TODO issue there to add a way to distinguish those cases because it seems like there's a good chance of a bug being created from that somewhere.